src/js/date.js

```diff
@@ -95,3 +100,41 @@ export const getNumericWeekday = (locale = 'en-uk', date = new Date()) => {
   });
   return weekdayIndex;
 };
+
+export const getLocaleDayMonthYear = (locale = 'en-uk', date = new Date()) => {
+  const objDate = date.toLocaleString(locale, { day: 'numeric', month: 'numeric', year: 'numeric' });
+  return objDate;
+};
+
+export const isDateValid = (locale = 'en-uk', inputValue = '') => {
+  // year, monthIndex, day
+  const blueprint = new Date(2017, 2, 23);
+
+  if (!Intl.DateTimeFormat.supportedLocalesOf(locale).length) {
+    throw new Error(`locale not supported: ${locale}`);
+  }
+
+  // find out which out of 4 valid seperator the current locale has
+  const localisedBlueprintDate = new Intl.DateTimeFormat(locale).format(blueprint);
+  const localisedBlueprintDateString = localisedBlueprintDate.toString();
+
+  const usedSeperator = localisedBlueprintDateString.match(ALL_DATE_SEPERATORS);
+
+  // find out how the locale date is structured (YYYY-MM-DD, YYYY-DD-MM, etc) using the blueprint
+  const splittedValue = inputValue.split(usedSeperator);
+  const splittedBlueprint = localisedBlueprintDateString.split(usedSeperator);
+
+  // we know month is 3 cause we set 2 in the date creation. In the creation it take 2 as monthIndex and
+  // in reading gives the actual month (index + 1)
+  const monthIndex = splittedBlueprint.indexOf('3');
```

Comment:
Looks good to me.
One last question, since we are actively parsing the date here, I would suggest to put it into it's own `parseLocalisedDate` function (however you name it)

Reply:
the method returns a parse date already. maybe i should change the name